File: expo-app/modules/expo-presage-emotion/ios/ExpoPresageEmotionModule.swift
import ExpoModulesCore

public class ExpoPresageEmotionModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoPresageEmotion")

    View(ExpoPresageEmotionView.self) {
      Events("onVitals")
      Prop("apiKey") { (_: ExpoPresageEmotionView, _: String?) in
        // No-op on iOS; Presage SmartSpectra is Android-only in this module
      }
    }
  }
}

File: expo-app/modules/expo-presage-emotion/android/src/main/java/expo/modules/presageemotion/ExpoPresageEmotionModule.kt
package expo.modules.presageemotion

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoPresageEmotionModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoPresageEmotion")

    View(ExpoPresageEmotionView::class) {
      Events("onVitals")

      Prop("apiKey") { view: ExpoPresageEmotionView, apiKey: String? ->
        view.setApiKey(apiKey ?: "")
      }
    }
  }
}
